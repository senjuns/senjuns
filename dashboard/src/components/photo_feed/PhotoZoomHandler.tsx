import IconButton from '@material-ui/core/IconButton';
import { makeStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import RemoveIcon from '@material-ui/icons/Remove';
import clsx from 'clsx';
import { FC } from 'react';

import { Size } from '../../shared/interfaces';
import { ZOOM_LEVELS } from './constants';

const useStyles = makeStyles((theme) => ({
  controls: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
  },
  zoomButton: {
    height: theme.spacing(5),
    width: theme.spacing(5),
    background: 'white',
    '&:hover': {
      background: 'white',
    },
    '&[disabled]': {
      background: 'white',
    },
    borderRadius: theme.spacing(1.5),
  },
  zoomIn: {
    marginBottom: theme.spacing(0.5),
  },
}));

interface PhotoZoomHandlerProps {
  /**
   * When the zoom out is disabled
   */
  isZoomOutDisabled?: boolean;
  /**
   * The zoom scale information of the Main Window.
   */
  scale: number;
  /**
   * The size of presentation layer.
   */
  presentationSize: Size;
  /**
   * Callback to be called when the scale delta changes.
   */
  onChangeScaleDelta: (delta: number) => void;
}

const PhotoZoomHandler: FC<PhotoZoomHandlerProps> = ({
  isZoomOutDisabled,
  scale,
  presentationSize,
  onChangeScaleDelta,
}) => {
  const classes = useStyles();

  const handleZoomIn = () => {
    onChangeScaleDelta(ZOOM_LEVELS.MAX);
  };

  const handleZoomOut = () => {
    onChangeScaleDelta(ZOOM_LEVELS.MIN);
  };

  return (
    <div
      id="photo-zoom-button-group"
      data-scale={scale}
      className={classes.controls}
      style={{
        top: presentationSize.height - 110,
        left: presentationSize.width - 60,
      }}
    >
      <IconButton
        id="photo-zoom-in-button"
        disabled={scale > 1}
        className={clsx(classes.zoomButton, classes.zoomIn)}
        onClick={handleZoomIn}
      >
        <AddIcon />
      </IconButton>
      <IconButton
        id="photo-zoom-out-button"
        disabled={scale < 1 || isZoomOutDisabled}
        className={classes.zoomButton}
        onClick={handleZoomOut}
      >
        <RemoveIcon />
      </IconButton>
    </div>
  );
};

export default PhotoZoomHandler;
