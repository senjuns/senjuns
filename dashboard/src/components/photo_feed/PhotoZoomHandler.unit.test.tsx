// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import PhotoZoomHandler from './PhotoZoomHandler';

describe('<PhotoZoomHandler />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PhotoZoomHandler
        isZoomOutDisabled={true}
        scale={1}
        presentationSize={{ height: 500, width: 500 }}
        onChangeScaleDelta={jest.fn()}
      />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
