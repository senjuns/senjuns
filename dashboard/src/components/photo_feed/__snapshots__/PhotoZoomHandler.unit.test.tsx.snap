// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoZoomHandler /> renders correctly. 1`] = `
<div
  className="makeStyles-controls-1"
  data-scale={1}
  id="photo-zoom-button-group"
  style={
    Object {
      "left": 440,
      "top": 390,
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-zoomButton-2 makeStyles-zoomIn-3"
    disabled={false}
    id="photo-zoom-in-button"
    onClick={[Function]}
  >
    <Memo />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-zoomButton-2"
    disabled={true}
    id="photo-zoom-out-button"
    onClick={[Function]}
  >
    <Memo />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
